### PR TITLE
WIP rebasing | feat(store-ui): Spinner atom

### PR DESCRIPTION
**Pull request by Gmantiqueira**
## What's the purpose of this pull request?
This PR is related to Loading Button component/task. IMO It doesn't make sense creating a molecule only for the Loading Button, since the loading is handled on the Button atom children. So, we just need a Spinner component to alternate between the Button text and a loading indicator.

### CODE REVIEW

File: packages/store-ui/src/atoms/Spinner/Spinner.test.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { render } from '@testing-library/react'
+import React from 'react'
+
+import Spinner from './Spinner'
+
+describe('Spinner', () => {
+  it('`data-store-spinner` is present', () => {
+    const { getByTestId } = render(<Spinner data-testid="loading" />)
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Spinner/Spinner.test.tsx:8]: What do you think about this suggestion? This follow the pattern of other components.
```suggestion
    const { getByTestId } = render(<Spinner data-testid="store-spinner" />)
```

File: packages/store-ui/src/index.ts [hunk 2]
```diff
@@ -35,6 +35,9 @@ export type { BadgeProps } from './atoms/Badge'
 export { default as Slider } from './atoms/Slider'
 export type { SliderProps } from './atoms/Slider'
 
+export { default as Spinner } from './atoms/Spinner'
```
Comment (igorbrasileiro) [inline packages/store-ui/src/index.ts:38]: Please pay attention to it.

File: packages/store-ui/src/atoms/Spinner/Spinner.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+const Spinner = forwardRef<HTMLDivElement, SpinnerProps>(function Spinner(
+  { children, testId = 'store-spinner', ...props },
+  ref
+) {
+  return (
+    <span ref={ref} data-store-spinner data-testid={testId} {...props}>
+      {children}
+    </span>
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Spinner/Spinner.tsx:17]: What do you think about this suggestion?
```suggestion
    <span ref={ref} data-store-spinner data-testid={testId} {...props} />
```

**Review by igorbrasileiro — COMMENTED** [review]
Nice job, @Gmantiqueira!
Can you left a store PR for us to see the performance impact?
Also, we should discuss only have the spinner and not the button with loading.

**Comment by Gmantiqueira** [comment]
created another PR https://github.com/vtex/faststore/pull/961

**Pull request closed without merging**